### PR TITLE
[FIX] travis_weblate.py: Adding more patch to de odoo

### PR DESCRIPTION
**Pull request by JesusZapata**
I ported the changes made in https://github.com/odoo/odoo/pull/17925 over the files `odoo/addons/base/ir/ir_translation.py` and `odoo/tools/translate.py`

Now MQT apply these patch if the patch has not been applied

The change in the file `(odoo|openerp)/addons/base/ir/ir_translation.py` 

image

The change in the file `(odoo|openerp)/tools/translate.py` 

image

### CODE REVIEW

**Comment by JesusZapata** [comment]
@moylop260 
I applied the changes made on https://github.com/odoo/odoo/pull/17925 over MQT

File: travis/travis_weblate.py [hunk 1]
```diff
@@ -67,17 +68,31 @@ def _apply_patch_odoo(self):
...
-                   r"s/translation'] = src/translation'] = ''/g",
-                   s_file]
-            print " ".join(cmd)
-            subprocess.call(cmd)
+        for base in ('odoo', 'openerp'):
+            p_file = os.path.join(self._server_path, base,
+                                  os.path.join('tools', 'translate.py'))
+            if os.path.isfile(p_file):
+                sed = ["sed", "-i", "-e",
+                       r"s/translation'] = src/translation'] = ''/g", p_file]
```
Comment (moylop260) [inline travis/travis_weblate.py]: You can avoid the loop `for base in ('odoo', 'openerp')` using directly `base = "{odoo,openerp}"` to cover both folders.

File: travis/travis_weblate.py [hunk 2]
```diff
@@ -67,17 +68,31 @@ def _apply_patch_odoo(self):
...
+                print " ".join(sed)
+                subprocess.call(sed)
+            p_file = os.path.join(self._server_path, base,
+                                  os.path.join('addons', 'base', 'ir',
+                                               'ir_translation_test.py'))
+            if os.path.isfile(p_file):
+                rgrep = ["rgrep", "-h", "-n", "'if not trans_dict'",
+                         p_file]
+                print " ".join(rgrep)
+                p = subprocess.Popen(shlex.split(" ".join(rgrep)),
```
Comment (moylop260) [inline travis/travis_weblate.py]: Use directly rgrep without split join

File: travis/travis_weblate.py [hunk 3]
```diff
@@ -67,17 +68,31 @@ def _apply_patch_odoo(self):
...
+            if os.path.isfile(p_file):
+                sed = ["sed", "-i", "-e",
+                       r"s/translation'] = src/translation'] = ''/g", p_file]
+                print " ".join(sed)
+                subprocess.call(sed)
+            p_file = os.path.join(self._server_path, base,
+                                  os.path.join('addons', 'base', 'ir',
+                                               'ir_translation_test.py'))
+            if os.path.isfile(p_file):
+                rgrep = ["rgrep", "-h", "-n", "'if not trans_dict'",
```
Comment (moylop260) [inline travis/travis_weblate.py]: You can use `sed -i '/pattern/d'` to delete a line and avoid a rgrep

File: travis/travis_weblate.py (hunk 1 above)
Comment (JesusZapata) [inline travis/travis_weblate.py]: @moylop260 
Thanks for teach me this way to use multiple folders in `rgrep` command
I try to use this example into the python and i can not made to work

image

I the correct way to run this command ?

**Comment by moylop260** [comment]
We need create a full POC with this change...
If there is a change to create we will use a new PR

**Pull request merged**